apps/api/src/app/controllers/__tests__/fetchManyViewModels.e2e.spec.ts

```diff
@@ -44,90 +44,93 @@ describe('When fetching multiple resources', () => {
-                        })
-                    );
+                        expect(res.status).toBe(httpStatusCodes.ok);
 
-                beforeEach(async () => {
-                    await testRepositoryProvider.addResourcesOfSingleType(ResourceType, [
-                        ...unpublishedResourcesToAdd,
-                        ...publishedResourcesToAdd,
-                    ]);
+                        expect(res.body.length).toBe(
+                            testDataWithAllResourcesPublished[ResourceType].length
+                        );
 
-                    await testRepositoryProvider.getTagRepository().createMany(tagTestData);
+                        expect(res.body).toMatchSnapshot();
+                    });
                 });
 
-                it('should return the expected number of results', async () => {
-                    const res = await request(app.getHttpServer()).get(
-                        `/resources${endpointUnderTest}`
+                describe(`when some of the resources are unpublished`, () => {
+                    /**
+                     * Note that there is no requirement that the test data have
+                     * `published = true`
+                     */
+                    const publishedResourcesToAdd = resourceTestData[ResourceType].map(
+                        (instance: Resource) =>
+                            instance.clone({
+                                published: true,
+                            })
                     );
 
-                    expect(res.body.length).toBe(publishedResourcesToAdd.length);
-
-                    // Sanity check
-                    expect(publishedResourcesToAdd.length).not.toEqual(
-                        unpublishedResourcesToAdd.length
-                    );
-
-                    expect(res.body).toMatchSnapshot();
+                    const unpublishedResourcesToAdd = resourceTestData[ResourceType]
+                        // We want a different number of published \ unpublished terms
+                        .slice(0, -1)
+                        .map((instance, index) =>
+                            instance.clone({
+                                id: `UNPUBLISHED-00${index + 1}`,
+                                published: false,
+                            })
+                        );
+
+                    beforeEach(async () => {
+                        await testRepositoryProvider.addResourcesOfSingleType(ResourceType, [
+                            ...unpublishedResourcesToAdd,
+                            ...publishedResourcesToAdd,
+                        ]);
+
+                        await testRepositoryProvider.getTagRepository().createMany(tagTestData);
+                    });
+
+                    it('should return the expected number of results', async () => {
+                        const res = await request(app.getHttpServer()).get(
+                            `/resources${endpointUnderTest}`
```

Comment:
just a detail but did we prefer having the slash here? `/resources/${endpointUnderTest}`

Reply:
I prefer it this way, personally.